### PR TITLE
IBP-4152-FieldMapChanges

### PR DESCRIPTION
**Pull request by gelli-leafnode**
Modify SQL Scripts, if blockId is not present (for uploaded via datasetimport and design import) 
If BlockId is not present but labels is not empty: default values will be assign for rowsperplot, machinecapicity and planting order this data are needed to display the plot 
Fix failing unit test

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/domain/fieldbook/FieldMapTrialInstanceInfo.java [hunk 1]
```diff
@@ -103,6 +103,10 @@
 	/** The deleted plot coordintes in (row, range) format */
 	private List<String> deletedPlots;
 
+	private boolean inValidValue;
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/domain/fieldbook/FieldMapTrialInstanceInfo.java:106]: Rename to `hasInvalidValue`

File: src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java [hunk 2]
```diff
@@ -1342,4 +1344,27 @@ void setDataSetBuilder(final DataSetBuilder dataSetBuilder) {
...
+
+	private boolean hasInvalidValues(List<FieldMapLabel> labels) {
+		if (!CollectionUtils.isEmpty(labels)) {
+			List<FieldMapLabel> invalid = labels.stream().filter(fieldMapLabel -> Util.getIntValue(fieldMapLabel.getColumn()) <= 0 || Util.getIntValue(fieldMapLabel.getRange()) <= 0).collect(Collectors.toList());
+			return !CollectionUtils.isEmpty(invalid);
+		}
+		return false;
+	}
+
+	private boolean hasOverlapping(List<FieldMapLabel> labels) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java:1356]: Rename to `hasOverlappingCoordinates`

File: src/main/java/org/generationcp/middleware/domain/fieldbook/FieldMapTrialInstanceInfo.java [hunk 3]
```diff
@@ -103,6 +103,10 @@
...
+
+	private boolean overlappingCoordinates;
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/domain/fieldbook/FieldMapTrialInstanceInfo.java:108]: Rename to `hasOverlappingCoordinates`

File: src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java [hunk 4]
```diff
@@ -1318,4 +1344,27 @@ void setDataSetBuilder(final DataSetBuilder dataSetBuilder) {
 	void setTrialEnvironmentBuilder(final TrialEnvironmentBuilder trialEnvironmentBuilder) {
 		this.trialEnvironmentBuilder = trialEnvironmentBuilder;
 	}
+
+	private boolean hasInvalidValues(List<FieldMapLabel> labels) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java:1348]: Rename to `hasInvalidCoordinateValue`

Comment (darla-leafnode) [reply src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java:1348]: can be simplied to one line:
return labels.stream().anyMatch(fieldMapLabel -> Util.getIntValue(fieldMapLabel.getColumn()) <= 0 || Util.getIntValue(fieldMapLabel.getRange()) <= 0);

File: src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java (hunk 2 above)
Comment (darla-leafnode) [reply src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java:1356]: No need for `hasOverlapping` variable. Immediately return true once a duplicate is found. return false at the end (if no dupe found)

File: src/main/java/org/generationcp/middleware/util/Util.java [hunk 5]
```diff
@@ -451,4 +451,11 @@ public static boolean isLeapYear(final int year) {
 		return Optional.absent();
 	}
 
+	public static int getIntValue(final Integer value) {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/util/Util.java:454]: No need for this Util method

File: src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java (hunk 4 above)
Comment (darla-leafnode) [inline src/main/java/org/generationcp/middleware/manager/StudyDataManagerImpl.java:1348]: Apply code cleanup to this class

File: src/main/java/org/generationcp/middleware/util/Util.java (hunk 5 above)
Comment (gelli-leafnode) [reply src/main/java/org/generationcp/middleware/util/Util.java:454]: Util.getIntValue currently used in StudyDataManagerImpl.updateFieldMapTrialInstanceInfo streams 

```
final List<FieldMapLabel> rows = trial.getFieldMapLabels().stream().filter(fieldMapLabel -> Util.getIntValue(fieldMapLabel.getColumn()) > 0).collect(
						Collectors.toList());
					final List<FieldMapLabel> ranges = trial.getFieldMapLabels().stream().filter(fieldMapLabel -> Util.getIntValue(fieldMapLabel.getRange()) > 0).collect(
						Collectors.toList());
```

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**